jwst/skymatch/skymatch_step.py

```diff
@@ -103,51 +89,44 @@ def process(self, input):
             binwidth=self.binwidth
         )
 
-        # group images by their "group id":
-        grp_img = img.models_grouped
-
-        # create a list of "Sky" Images and/or Groups:
         images = []
-        grp_id = 1
-
-        for g in grp_img:
-            if len(g) > 1:
-                images.append(
-                    SkyGroup(
-                        list(map(self._imodel2skyim, g)),
-                        id=grp_id
-                    )
-                )
-                grp_id += 1
-            elif len(g) == 1:
-                images.append(self._imodel2skyim(g[0]))
-            else:
-                raise AssertionError("Logical error in the pipeline code.")
+        with library:
+            for group_index, (group_id, group_inds) in enumerate(library.group_indices.items()):
+                sky_images = []
+                for index in group_inds:
+                    model = library.borrow(index)
+                    sky_images.append(self._imodel2skyim(model, index))
+                    library.shelve(model, index, modify=False)
+                if len(sky_images) == 1:
+                    images.extend(sky_images)
+                else:
+                    # FIXME: why does this use a number for group_index?
```

Comment:
I think it was me that added this `FIXME`. It makes sense to me to leave this to a follow-up PR. I opened an issue to track this in case removing this `FIXME` is preferred:
https://github.com/spacetelescope/jwst/issues/8711

Reply:
I don't really care whether the FIXME is there or not, but I guess if you already made the issue, then the comment can be safely removed.  Thanks for doing that.